dash/dash.py

```diff
@@ -105,7 +105,10 @@ def __init__(
         self.config = _AttributeDict({
             'suppress_callback_exceptions': False,
             'routes_pathname_prefix': url_base_pathname,
-            'requests_pathname_prefix': url_base_pathname,
+            'requests_pathname_prefix': os.getenv(
+                'DASH_REQUESTS_PATHNAME_PREFIX',
+                '/{}/'.format(os.environ['DASH_APP_NAME'])
+                if 'DASH_APP_NAME' in os.environ else url_base_pathname),
```

Comment:
It looks like this makes the os.environ parameter take precedence over the value of `url_base_pathname` set in the constructor, which might go against @chriddyp's suggestion in #312 of having config values set in the constructor take precedence. (Although the fallback logic here is a bit more subtle than simple parameter shadowing). But then there's the challenge of detecting whether a user specified a value through the constructor or if the value is just the default. I'm not sure what the best of doing that is though.

I wonder if it might make sense to come up with a systematic abstraction for getting config values from the constructor, falling back to os.environ (and handling either upper or lower case DASH_ vars), and then falling back to a final default value? This is probably taking the discussion further afield of this PR (sorry!) but one possibility is to switch to using `**kwargs` in the constructor. eg:

`some_param = kwargs.get('some_param', os.environ.get('dash_some_param', os.environ.get('DASH_SOME_PARAM', defaults.some_param)))`

Although I suppose that breaks IDE autocomplete `for Dash.__init__`...

(apologies if I'm missing some additional context to this PR also!)